Topics/Software_Tools/PyAutoGUI/Automation_with_PyAutoGUI.md

```diff
@@ -0,0 +1,140 @@
+print(pyautogui.position()) # prints the current position of the mouse
+```
+All you have to do is simple run the program above and move your mouse to the location you want the coordinates for (make sure its within 5 seconds). The output will tell you exactly where the mouse is!
+
+### Clicking
+Now that you can move your mouse around freely to wherever you want, how do you finally perform a mouseclick? PyAutoGUI provides multiple options as follows:
+```python
+>>> pyautogui.click()  # perform a simple left-click
+>>> pyautogui.click(x=450, y=230)  # move to 450, 230, then left-click
+>>> pyautogui.click(button='right')  # right-click the mouse
+>>> pyautogui.doubleClick()  # double left-click
+>>> pyautogui.click(clicks=3)  # triple left-click
+>>> pyautogui.click(button='right', clicks=4, interval=0.5) #quadruple right-click 
+# but with a 0.5 second pause in between the clicks
+```
+
+### Other Cool Mouse Functions
+There's a lot more than just clicking that you can do with the PyAutoGUI mouse.
+Here are a few examples:
+```python
+>>> pyautogui.dragTo(300, 400, 2, button='left')  # drag mouse to X of 300, Y of 400 over 2 seconds while holding down left mouse button
+>>> pyautogui.drag(30, 0, 2, button='right')   # drag the mouse left 30 pixels over 2 seconds while holding down the right mouse button
+>>> pyautogui.scroll(10)   # scroll up 10 "clicks"
+>>> pyautogui.hscroll(-10)  # scroll left 10 "clicks"
+```
+Go check out the docs for a lot more!
+
+## Keyboard Controls
+Pending
+
+## Locating Images and Screenshots
+Let's say you want to click on something that's going to pop up on the screen but you don't know where it's going to be. PyAutoGUI has just the solution for that. 
+
+Suppose you want to open the UofT folder on your desktop below, but you don't want to bother finding its coordinates.
+![desktop](./assets/desktop.png)
+
+First take a screenshot of the folder and save it:
+
+![desktop](./assets/uoft_folder.png)
+
+You have two options:
+- Locate the center of the folder and click on it
+```python
+>>> import pyautogui
+>>> uoft_folder = pyautogui.locateOnScreen('uoft_folder.png')
+>>> uoft_mid = pyautogui.center(uoft_folder)
+>>> uoft_midx, uoft_midy = uoft_mid
+>>> pyautogui.click(uoft_midx, uoft_midy, clicks=2)
+```
+
+- Or you could take advantage of the PyAutoGUI shorthands
+```python
+>>> import pyautogui
+>>> pyautogui.click('uoft_folder.png') # yes it's that easy
+```
+
+Additionally, PyAutoGUI also provides screenshot functionality, in case you need it.
+```python
+>>> import pyautogui
+# Takes a screenshot and returns an PIL Image object
+>>> im1 = pyautogui.screenshot()
+# This one also saves the screenshot with the name being the string passed in
+>>> im2 = pyautogui.screenshot('my_screenshot.png')
+# You can also take screenshots of specific regions by specifying the x,y coordinates of the top left and bottom right of the desired region
+>>> im3 = pyautogui.screenshot(region=(0,0, 300, 400))
```

Comment:
It's great that you've provided multiple methods for capturing screenshots using PyAutoGui, showcasing versatility and accommodating different user preferences or requirements. This approach not only enriches the tutorial but also empowers readers with various options to accomplish their tasks efficiently.

Reply:
Thank you for the warm feedback!